cmd/clusterctl/client/cluster/installer.go

```diff
@@ -103,7 +103,11 @@ func (i *providerInstaller) Install(ctx context.Context, opts InstallOptions) ([
 		ret = append(ret, components)
 	}
 
-	return ret, waitForProvidersReady(ctx, opts, i.installQueue, i.proxy)
+	if err := waitForProvidersReady(ctx, opts, i.installQueue, i.proxy); err != nil {
```

Comment:
We intentionally added a flag to make wait for providers opt-in. It feels wrong to add waitForCAInjection for all cases.

Does it make sense to waitForCAInjection when waitForProviders is disabled? (this is the default)

Maybe waitForCAInjection should simplyi be part of waitForProvidersReady

If not, we problably would want to add another flag and disable per default (we can still had-code it to true in the tests / test framework)